scanpipe/models.py

```diff
@@ -2141,6 +2142,19 @@ def is_symlink(self):
         """Return True, if the resource is a symlink."""
         return self.type == self.Type.SYMLINK
 
+    @property
+    def is_legal(self):
```

Comment:
Missing a unit test for this.
Also, is `or base_name.startswith(LEGAL_STARTS_ENDS)` needed? Is the start of `base_name` ever different from the start of `name`? If that's a possibility, make sure to cover it in the test.

Reply:
Added unit test.
removed `base_name.startswith(LEGAL_STARTS_ENDS)`